### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_static-list-of-todos/)

### CODE REVIEW

File: src/components/UserInfo/UserInfo.tsx [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
-// Create a `UserInfo` component accepting a `user` object and use it to render
-// a `todo.user` in the list with some styling. (Show at least a `name` and an
-// `email` of the `user`)
+type User = {
+  name: string,
+  email: string
+};
 
-// Add a default export statement for UserInfo component to use it in the other files
+export const UserInfo: React.FC <User> = ({
```
Comment (Anna-Tkachenko) [inline src/components/UserInfo/UserInfo.tsx:8]: UserInfo component should accept full user object
<UserInfo user={user} />

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
 
-// Add a default export statement for TodoInfo component to use it in the other files
+interface Todo {
+  userId: number,
+  id: number,
+  title: string,
+  completed: boolean,
+  user: User | undefined;
+}
+
```
Comment (Anna-Tkachenko) [inline src/components/TodoList/TodoList.tsx:16]: Create separate file for all types and export them from it
You shouldn't create the same types in each component

File: src/components/TodoInfo/TodoInfo.tsx [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
-// Add a default export statement for TodoInfo component to use it in the other files
+interface Todo {
+  userId: number,
+  id: number,
+  title: string | undefined,
+  completed: boolean,
+  user: User | undefined;
+}
+
+export const TodoInfo: React.FC <Todo> = ({
```
Comment (Anna-Tkachenko) [inline src/components/TodoInfo/TodoInfo.tsx:17]: TodoInfo component should accept full todo object
<TodoInfo todo={todo} />

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
-// Create a `TodoList` component accepting an array of `preparedTodos` and
-// rendering them as a list
+import { Props } from '../../types';
 
-// Add a default export statement for TodoInfo component to use it in the other files
+export const TodoList: React.FC <Props> = ({ todos }) => (
+  <ul className="list container__list">
+    {todos.map(todo => (
+      <li key={todo.id} className="list__item">
+        <TodoInfo {...todo} />
```
Comment (Anna-Tkachenko) [inline src/components/TodoList/TodoList.tsx:10]: TodoInfo component should accept full object in separate property
```suggestion
        <TodoInfo todo={todo} />
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 5]
```diff
@@ -1,6 +1,31 @@
...
+  title,
+  completed,
+  user,
+}) => {
+  return (
+    <>
+      {user && (<UserInfo {...user} />)}
```
Comment (Anna-Tkachenko) [inline src/components/TodoInfo/TodoInfo.tsx:24]: UserInfo component should accept full user object
```suggestion
      {user && (<UserInfo user={user} />)}
```

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -1,13 +1,29 @@
...
 
 const App: React.FC = () => (
-  <div className="App">
-    <h1>Static list of todos</h1>
-  </div>
+  <>
+    <div className="container">
+      <TodoList todos={preparedTodos} />
+    </div>
+  </>
```
Comment (Denchepornyuk) [inline src/App.tsx:26]: do you need react.fragment here?

File: src/components/UserInfo/UserInfo.tsx [hunk 7]
```diff
@@ -1,7 +1,22 @@
...
-// Add a default export statement for UserInfo component to use it in the other files
+export const UserInfo: React.FC <UserProps> = ({ user }) => {
+  const { name, email } = user;
+
+  return (
+    <>
+      <h2 className="list__item-username">
+        Name:
+        {' '}
+        {name}
```
Comment (Denchepornyuk) [inline src/components/UserInfo/UserInfo.tsx:13]:
```suggestion
        {`Name: ${name}`}
```

File: src/components/UserInfo/UserInfo.tsx [hunk 8]
```diff
@@ -1,7 +1,22 @@
...
+      </h2>
+      <h3 className="list__item-email">
+        Email:
+        {' '}
+        {email}
```
Comment (Denchepornyuk) [inline src/components/UserInfo/UserInfo.tsx:18]:
```suggestion
        {`Email: ${email}`}
```

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**